### PR TITLE
Miscellaneous Modifications

### PR DESCRIPTION
**Pull request by ferricoxide**
* Per out-of-band conversation: add ability to launch service-duplicates via use of RDS snapshots
* Add usage documentation/guidance
* Make invocation of site's `plugins.sh` more forgiving (improves chances that all valid plugins will load even if some plugins have become incompatible over time)
* Turn of provisioning-time SCAP scanning (to speed up deployment)
* Allow creation of RDS elements with "friendlier" naming
* Update instance role to support SSM and CloudWatch
* Improve setup of cron-jobs to work better with SCL's Ruby packaging

### CODE REVIEW

File: Templates/make_RedMine_SGs.tmplt.json [hunk 1]
```diff
@@ -52,15 +53,52 @@
...
+        "ToPort": "111"
+      }
+    },
+    "UpdateNasSgRpcbindUdp": {
+      "DependsOn": "NasSg",
+      "Type": "AWS::EC2::SecurityGroupIngress",
+      "Properties": {
+        "Description": "Exceptions for rpcbind over UDP",
+        "GroupId": { "Ref": "NasSg" },
+        "SourceSecurityGroupId": { "Ref": "NasSg" },
```
Comment (lorengordon) [inline Templates/make_RedMine_SGs.tmplt.json:74]: Is `NasSg` attached to both the redmine EC2 instance and the EFS volume?

Comment (ferricoxide) [reply Templates/make_RedMine_SGs.tmplt.json:74]: The EC2 instance has all three SGs. Each of the SGs lassos the EC2 and a given service (i.e., EC2/EFS; EC2/ELB; EC2/RDS)

Comment (lorengordon) [reply Templates/make_RedMine_SGs.tmplt.json:74]: Got it, just making sure, cuz otherwise this wouldn't work. 😆

File: docs/Task-Upgrades.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+Changing to a newer AMI (via the tempate's `AmiId` parameter-value) forces an instance-replacement. This will change the starting RPM-set of the deployed instance. Additionally, the included hardening tasks perform a `yum update` action
+
+### RDS Engine Upgrades
+
+RDS is an Amazon-managed service. Upgrades happen automatically and transparently. There is not currently anything for a CFn-user to do.
+
+## Notes and Caveats
+
+* Wheter deployed via a "parent" template or individual element-templates, updates _can_ be applied via the individual element-template. If deployed via a "parent" template, direct updates of element-level templates is highly-discouraged. If attempting such an update from within the CloudFormation web console, an alert will pop up warning against such an action. Heed that warning!
```
Comment (lorengordon) [inline docs/Task-Upgrades.md]: Wheter -> Whether

File: docs/Task-Upgrades.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+Re-running can be done interactively on the already deployed instance or by forcing an instance-replacement. It is highly discouraged to do the former, however. To force an instance-replacement, execute a stack-update action, changing the curret value of the `ToggleNewInstances` parameter.
+
+### Operating System Upgrades
+
+Operating system updates can take two forms: a `yum update` type of operation or an AMI update operation.
+
+#### Yum Update Method
+
+Each time a new EC2 is launched, the included hardening tasks perform a `yum update` action. Force an instance-replacement by executing a stack-update action with a change of the curret value of the `ToggleNewInstances` parameter.
```
Comment (lorengordon) [inline docs/Task-Upgrades.md]: curret -> current

File: docs/Task-Upgrades.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+  - Between-AMI-publications OS patches
+- Upgrades to the RDS engine
+
+In general, upgrades are expected to be handled via a "stack update" action. When performing a stack-update, modify the relevant parameters' values to force the relevant update.
+
+### RedMine Application Upgrades
+
+To upgrade the version of RedMine deployed, update the `RedmineBinaryVersion` parameter's version-string. This string typically takes the form `redmine-<X.Y.Z>`.
+
+Note: current functioning of the project's `redmine-appinstall.sh` script takes the `RedmineBinaryVersion` parameter's string-value, appends it to the standard RedMine download source, `http://www.redmine.org/releases/` and then fetches the software from that assembled URL. If installing into a VPC that does not have access to the Internet-hosted RedMine download site, it will be nessary to alter the root URL. The download repository URL is easily modifiable in the script's header section.
```
Comment (lorengordon) [inline docs/Task-Upgrades.md]: nessary -> necessary

File: docs/Task-Deployment.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+#### Snapshot-based deployment
+
+If deploying from snapshot, uderstand the following limitations:
+
+* Changing the "Master User" attribute is not possible. When the "snapshot ID" parameter is set, the "Master User" value is ignored by the RDS creation-tasks. Attempts to subsequently alter the master user's name (by way of stack update actions) will result in data the data from the snapshot being lost.
+* Changing the master user's "Password" attribute is not possible. The templated RDS creation-processes ignore the attempt to change.
+* If using the "parent" template to orchestrate, it will be necessary to supply the RDS credentials from the snapshot source RDS. These credntials will be propagated into the resultant EC2 instance. If credentials are not supplied or do not match what is within the snapshot, the resultant EC2 instance(s) will not have the correct credentials to work with the database.
```
Comment (lorengordon) [inline docs/Task-Deployment.md]: credntials -> credentials

File: docs/Task-Deployment.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+1. _Execute the EFS service setup:_ A generic, three-AZ EFS share is configured. The "storage" security group is applied to allow NFS traffic only from the RedMine EC2 host.
+1. _Execute the ELB service setup:_ An application load-balancer is configured. The "application" security group is applied to all HTTPS-based requests from any source and allow HTTP-based communications only between the active RedMine EC2 instance and the load-balancer. A user-selected, ACM-managed SSL certificate is bound to the load-balancer.
+1. _Execute the IAM role setup:_ An IAM policy and attachable instance-role is created. The IAM objects allow unfettered access to the to the S3 bucket from the RedMine-hosting EC2 instances for the purposes of attachment backups. The IAM policy also allows: the AutoScaling service to launch and terminate EC2 instances the the ASG-managed EC2 instaces to use CloudWatch agent for log-export/centralized-logging; and, control of the EC2s via AWS Systems Manager.
```
Comment (lorengordon) [inline docs/Task-Deployment.md]: extraneous colon? `allows:`

"the the" -> "the"

Actually, that last sentence just reads kind of weird. Not sure how to edit.

File: docs/Task-Deployment.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+* S3 buckets
+* EC2 security groups
+* EC2 load balancers
+* EFS storage services
+* RDS database clusters and instances
+* EC2 autoscaling groups, launch configurations and instances
+
+If all of these privileges are not present in a single user/role, it will be necessary to execute the templates individually with users/roles appropriate to the individual template.
+
+Whether running via the inividual CFn templates or the "parent" template, the general deployment process/flow looks like:
```
Comment (lorengordon) [inline docs/Task-Deployment.md]: inividual -> individual

File: docs/Task-Deployment.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
+## Initial Deployment
+
+This project provides a collection of CloudFormation (CFn) templates to handle the deployment of all of the AWS components necessary to support a RedMine deployment. Additionally, this project includes a "parent" template. The per-service templates can be run individually or their running can handled in a monolithic, coordinated fashion via the parent template. 
+
+It will be necessary for the templates' user to have sufficient privileges to do configure all of the AWS elements present in this template-set. This means the user will need to have privileges to create/modify:
```
Comment (lorengordon) [inline docs/Task-Deployment.md]: "to do configure" -> "to configure"?

File: docs/Task-Deployment.md [hunk 9]
```diff
@@ -0,0 +1,45 @@
+## Initial Deployment
+
+This project provides a collection of CloudFormation (CFn) templates to handle the deployment of all of the AWS components necessary to support a RedMine deployment. Additionally, this project includes a "parent" template. The per-service templates can be run individually or their running can handled in a monolithic, coordinated fashion via the parent template.
```
Comment (lorengordon) [inline docs/Task-Deployment.md]: "can handled" -> "can be handled"

File: docs/Task-Deployment.md (hunk 8 above)
Comment (ferricoxide) [reply docs/Task-Deployment.md]: Think I meant more like "to execute"

File: docs/Task-Deployment.md [hunk 10]
```diff
@@ -22,7 +22,7 @@ Whether running via the inividual CFn templates or the "parent" template, the ge
...
-1. _Execute the IAM role setup:_ An IAM policy and attachable instance-role is created. The IAM objects allow unfettered access to the to the S3 bucket from the RedMine-hosting EC2 instances for the purposes of attachment backups. The IAM policy also allows: the AutoScaling service to launch and terminate EC2 instances the the ASG-managed EC2 instaces to use CloudWatch agent for log-export/centralized-logging; and, control of the EC2s via AWS Systems Manager.
+1. _Execute the IAM role setup:_ An IAM policy and attachable instance-role is created. The IAM objects allow unfettered access to the to the S3 bucket from the RedMine-hosting EC2 instances for the purposes of attachment backups. The IAM policy also allows the AutoScaling service to launch and terminate EC2 instances the the ASG-managed EC2 instaces to use CloudWatch agent for log-export/centralized-logging; and, control of the EC2s via AWS Systems Manager.
```
Comment (lorengordon) [inline docs/Task-Deployment.md]: I'm still struggling with this sentence:

> The IAM policy also allows the AutoScaling service to launch and terminate EC2 instances the the ASG-managed EC2 instaces to use CloudWatch agent for log-export/centralized-logging; and, control of the EC2s via AWS Systems Manager.

Comment (ferricoxide) [reply docs/Task-Deployment.md]: My keyboard's been wonky, lately (at over 2.5 years old, it's lasted longer than any laptop keyboard I've ever had). I think something got zapped.

**Review by lorengordon — APPROVED** [review]
Merge away

**Pull request merged**